misk/testing/src/main/kotlin/misk/testing/MiskTest.kt

```diff
@@ -0,0 +1,23 @@
+package misk.testing
+
+import com.google.inject.Module
+import org.junit.jupiter.api.extension.ExtendWith
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.CLASS)
+@ExtendWith(MiskTestExtension::class)
+/**
+ * Annotate your test classes with `@MiskTest` to have fields annotated with `@Inject` initialized.
+ * Provide zero or more [Module] classes or [Module] instances to be installed by annotating a
+ * [ModuleProvider] field in your test class with [Modules].
+ */
+annotation class MiskTest
+
+@Target(AnnotationTarget.FIELD)
+annotation class Modules
+
+data class ModuleProvider(val moduleList: List<Module> = ArrayList(), val moduleClassList: List<KClass<out Module>> = ArrayList()) {
```

Comment:
This signature is kind of odd, do we want to just accept `Module` instances, so we don't split between classes and objects?

Reply:
This allows for a combination to be injected. Happy to remove the signature for now and just allow `ModuleProvider` to accept multiple classes _or_ multiple instances, and add the ability to do both later if needed.

Reply:
But also just as happy to keep it even simpler and just accept instances rather than classes.

Reply:
PTAL.